OpenRVS/classes/OpenBeacon.uc

```diff
@@ -7,16 +10,189 @@
+			szIPAddr = left( szIPAddr, InStr( szIPAddr, ":" ) );
+			iPingTimeMS[iNumPlayers] = aPC.PlayerReplicationInfo.Ping;
+			iKillCount[iNumPlayers] = aPC.PlayerReplicationInfo.m_iKillCount;
+			fPlayingTime[iNumPlayers] = GetPlayingTime( szIPAddr );
+			iNumPlayers++;
+		}
+	}
+	textData = textData @ PlayerTimeMarker $ " ";
+	for (iCounter = 0; iCounter < iNumPlayers; iCounter++ )
+	{
+		textData = textData $ "/" $ DisplayTime( INT( fPlayingTime[iCounter] ) );
+	}
+	textData = textData @ PlayerPingMarker $ " ";
+	for (iCounter = 0; iCounter < iNumPlayers; iCounter++ )
+	{
+		textData = textData $ "/" $ iPingTimeMS[iCounter];
+	}
+	textData = textData @ PlayerKillMarker $ " ";
+	for (iCounter = 0; iCounter < iNumPlayers; iCounter++ )
+	{
+		textData = textData $ "/" $ iKillCount[iCounter];
+	}
+	textData = textData @ NumPlayersMarker @ iNumPlayers;
+	textData = textData @ RoundsPerMatchMarker @ pServerOptions.RoundsPerMatch;
+	textData = textData @ RoundTimeMarker @ pServerOptions.RoundTime;
+	textData = textData @ BetTimeMarker @ pServerOptions.BetweenRoundTime;
+	if (pServerOptions.BombTime > -1)
+		textData = textData @ BombTimeMarker @ pServerOptions.BombTime;
+	if ( pServerOptions.ShowNames )
+		integerData = 1;
+	else
+		integerData = 0;
+	textData = textData @ ShowNamesMarker @ integerData;
+	if ( pServerOptions.InternetServer )
+		integerData = 1;
+	else
+		integerData = 0;
+	textData = textData @ InternetServerMarker @ integerData;
+	if ( pServerOptions.FriendlyFire )
+		integerData = 1;
+	else
+		integerData = 0;
+	textData = textData @ FriendlyFireMarker @ integerData;
+	if ( pServerOptions.Autobalance )
+		integerData = 1;
+	else
+		integerData = 0;
+	textData = textData @ AutoBalTeamMarker @ integerData;
+	if ( pServerOptions.TeamKillerPenalty )
+		integerData = 1;
+	else
+		integerData = 0;
+	textData = textData @ TKPenaltyMarker @ integerData;
+	textData = textData @ GameVersionMarker @ Level.GetGameVersion();
+	if ( pServerOptions.AllowRadar )
+		integerData = 1;
+	else
+		integerData = 0;
+	textData = textData @ AllowRadarMarker @ integerData;
+	textData = textData @ LobbyServerIDMarker $ " 0";
+	textData = textData @ GroupIDMarker $ " 0";
+	textData = textData @ BeaconPortMarker @ boundport;
+	textData = textData @ NumTerroMarker @ pServerOptions.NbTerro;
+	if ( pServerOptions.AIBkp )
+		integerData = 1;
+	else
+		integerData = 0;
+	textData = textData @ AIBkpMarker @ integerData;
+	if ( pServerOptions.RotateMap )
+		integerData = 1;
+	else
+		integerData = 0;
+	textData = textData @ RotateMapMarker @ integerData;
+	if ( pServerOptions.ForceFPersonWeapon )
+		integerData = 1;
+	else
+		integerData = 0;
+	textData = textData @ ForceFPWpnMarker @ integerData;
+	textData = textData @ ModNameMarker @ class'Actor'.static.GetModMgr().m_pCurrentMod.m_szKeyWord;
+	textData = textData @ PunkBusterMarker $ " 0";
+
+	// Begin OpenRVS modifications.
+	textData = textData @ MOTDMarker @ pServerOptions.MOTD;

```

Comment:
Should we put a max length?  Something like:
s = pServerOptions.MOTD;
if ( len(s) > 100 )
{
s = left(s,100);
}

or something like that?  Does MOTD have a default max length already?

Reply:
There's no default max in code, but in-game you can only see the first 60 characters of any MOTD. I'll set this to 60.